pallets/msa/src/tests.rs

```diff
@@ -233,6 +235,40 @@ fn it_revokes_msa_key_successfully() {
 	})
 }
 
+// newtest
+#[test]
+fn it_retires_msa_and_revokes_keys_successfully() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Msa::create(test_origin_signed(1)));
+		assert_ok!(Msa::add_key(2, &test_public(1), EMPTY_FUNCTION));
+
+		assert_ok!(Msa::retire_my_msa(test_origin_signed(1)));
+
+		let info = Msa::get_key_info(&test_public(1));
+		assert_eq!(info, Some(KeyInfo { msa_id: 1, expired: 1, nonce: 0 }));
+
+	})
+}
+
+// newtest
+#[test]
+fn it_retires_msa_and_removes_delegations_successfully() {
+	new_test_ext().execute_with(|| {
+
+		create_delegation_relationships(3);
+
+		assert_ok!(Msa::retire_my_msa(test_origin_signed(1)));
+
+		for i in 1..4 {
+			assert_eq!(
+				Msa::get_provider_info_of(Provider(i), Delegator(1)).unwrap(),
+				ProviderInfo { expired: 1, permission: 0 },
+			);
+		}
+
+	})
```

Comment:
I added the happy path tests, but I would like to figure out the error handling portion before pushing the rest of the tests